server/testkit/tests/identity_verification_tests.rs

```diff
@@ -0,0 +1,184 @@
+    assert_eq!(
+        response,
+        IdentifyUserResponse::IdentityVerificationUnavailable
+    )
+}
+
+#[kanidmd_testkit::test]
+async fn test_start_with_available_identity(rsclient: KanidmClient) {
+    setup_server(&rsclient).await;
+    create_valid_test_user(&rsclient).await;
+    let response = make_request(&rsclient, IdentifyUserRequest::Start, None).await;
+    assert_eq!(response, IdentifyUserResponse::WaitForCode)
+}
+
+#[kanidmd_testkit::test]
+#[allow(non_snake_case)]
+async fn test_SPNDisplayAndInsertion_with_invalid_spn(rsclient: KanidmClient) {
+    setup_server(&rsclient).await;
+    create_valid_test_user(&rsclient).await;
+    let response = make_request(
+        &rsclient,
+        IdentifyUserRequest::Start,
+        Some("aasda".to_string()),
+    )
+    .await;
+    assert_eq!(response, IdentifyUserResponse::InvalidUserId);
+}
+
+async fn test_SPNDisplayAndInsertion_with_valid_spn(rsclient: KanidmClient) {
+    setup_server(&rsclient).await;
+    create_valid_test_user(&rsclient).await;
+    let response = make_request(
+        &rsclient,
+        IdentifyUserRequest::Start,
+        Some("testaccount".to_string()),
+    )
+    .await;
+    assert!(matches!(response, IdentifyUserResponse::ProvideCode { .. }));
+}
+async fn setup_server(rsclient: &KanidmClient) {
+    // basically this function logs in
+    let res = rsclient
+        .auth_simple_password("admin", ADMIN_TEST_PASSWORD)
+        .await;
+    // To enable the admin to actually make some of these changes, we have
+    // to make them a people admin. NOT recommended in production!
+    rsclient
+        .idm_group_add_members("idm_people_account_password_import_priv", &["admin"])
+        .await
+        .unwrap();
+    rsclient
+        .idm_group_add_members("idm_admins", &["admin"])
+        .await
+        .unwrap();
+}
+
+async fn create_other_user_valid(rsclient: &KanidmClient) {
+    let e: Entry = serde_json::from_str(
+        r#"{
+            "attrs": {
+                "class": ["account", "service_account"],
+                "name": ["testaccount_2"],
+                "displayname": ["testaccount"],
+                "ECDHKey": [45,45,45,45,45,66,69,71,73,78,32,80,82,73,86,65,84,69,32,75,69,89,45,45,45,45,45,10,77,73,71,72,65,103,69,65,77,66,77,71,66,121,113,71,83,77,52,57,65,103,69,71,67,67,113,71,83,77,52,57,65,119,69,72,66,71,48,119,97,119,73,66,65,81,81,103,99,53,67,106,106,68,89,74,101,75,79,51,43,52,52,115,10,102,82,122,97,79,52,52,53,53,116,86,80,116,105,75,114,47,106,70,122,102,115,81,106,53,108,117,104,82,65,78,67,65,65,83,87,86,104,86,82,48,117,51,107,90,103,104,65,74,109,106,68,51,82,52,104,68,89,97,81,118,102,75,121,10,113,87,100,49,50,66,99,57,67,48,74,108,122,99,66,73,75,86,81,120,108,87,43,65,120,115,98,82,108,88,118,108,65,74,75,108,74,115,51,47,53,54,81,122,71,120,90,85,65,111,107,104,115,97,56,51,10,45,45,45,45,45,69,78,68,32,80,82,73,86,65,84,69,32,75,69,89,45,45,45,45,45,10],
```

Comment:
You'll never need to externally create the ec keys, the server will generate these internally.

Reply:
Also you should be making persons, not accounts. service accounts don't need to verify each other! humans do :)

Reply:
sure!